src/GNDStk/BodyText/src/types.hpp

```diff
@@ -0,0 +1,35 @@
+
+// -----------------------------------------------------------------------------
+// Variant<vectors> and variant<scalars> types
+// -----------------------------------------------------------------------------
+
+using VariantOfVectors = std::variant<
+   // strings
+   std::vector<std::string>,
+
+   // chars
+   std::vector<char>,
+
+   // integrals
+   // Remember that signed char != char in C++
+   std::vector<signed char>,
+   std::vector<short>,
+   std::vector<int>,
+   std::vector<long>,
+   std::vector<long long>,
+
+   // unsigned integrals
+   std::vector<unsigned char>,
+   std::vector<unsigned short>,
+   std::vector<unsigned int>,
+   std::vector<unsigned long>,
+   std::vector<unsigned long long>,
+
+   // floating-points
+   std::vector<float>,
+   std::vector<double>,
+   std::vector<long double>
```

Comment:
I don't think there's a need to support so many different types. string, int and double will be more than sufficient.

Reply:
Probably not, although it's easy to support them, and I'm inclined to think that an occasional user may actually prefer certain specific types other than string, int, and double. I'd want to support at least float and long double, in additional to double, as I can see scenarios where either of those might be warranted.